app/app/explore/page.tsx

```diff
@@ -0,0 +1,280 @@
+"use client";
+
+import { useState, useEffect } from 'react';
+import Image from 'next/image';
+import TranslucentCard from '../translucentCard';
+import { fetchJson, formatNumber } from '@/app/utils';
+
+interface Pool {
+  asset: string;
+  assetPriceUSD: string;
+  volume24h: string;
+  depth: string;
+  poolAPY: string;
+  assetDepth: string;
+  runeDepth: string;
+  nativeDecimal: string;
+}
+
+interface Saver {
+  asset: string;
+  saversCount: number;
+  saversReturn: string;
+  earned: string;
+  filled: number;
+  assetPriceUSD: string;
+  saversDepth: string;
+  assetDepth: string;
+  synthSupply: string;
+}
+
+const Explore = () => {
+  const [activeTab, setActiveTab] = useState('liquidity');
+  const [pools, setPools] = useState<Pool[]>([]);
+  const [savers, setSavers] = useState<Saver[]>([]);
+  const [runePriceUSD, setRunePriceUSD] = useState<number>(0);
+  const [loading, setLoading] = useState(true);
+  const [sortConfig, setSortConfig] = useState({ key: '', direction: '' });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const [poolsData, statsData, saversData] = await Promise.all([
+          fetchJson('https://midgard.ninerealms.com/v2/pools'),
+          fetchJson('https://midgard.ninerealms.com/v2/stats'),
+          fetchJson('https://vanaheimex.com/api/saversInfo')
+        ]);
+        setPools(poolsData);
+        setSavers(Object.values(saversData).map(s => s.savers));
+        setRunePriceUSD(parseFloat(statsData.runePriceUSD));
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const getAssetSymbol = (asset: string) => {
+    return asset.split(".")[1].split("-")[0] || asset;
+  };
+
+  const getLogoPath = (asset: string) => {
+    const assetLower = asset.toLowerCase();
+    return `https://storage.googleapis.com/token-list-swapkit-dev/images/${assetLower}.png`;
+  };
+
+  const calculateTVL = (pool: Pool) => {
+    const assetValueInUSD = parseFloat(pool.assetDepth) * parseFloat(pool.assetPriceUSD) / 1e8;
+    const runeValueInUSD = parseFloat(pool.runeDepth) * runePriceUSD / 1e8;
+    return (assetValueInUSD + runeValueInUSD) / 1e6;
+  };
+
+  const calculateVolumeUSD = (pool: Pool) => {
+    const volumeInRune = parseFloat(pool.volume24h) / 1e8;
+    return volumeInRune * runePriceUSD;
+  };
+
+  const calculateVolumeDepthRatio = (pool: Pool) => {
+    const volumeUSD = calculateVolumeUSD(pool);
+    const tvlUSD = calculateTVL(pool) * 1e6;
+    return volumeUSD / tvlUSD;
+  };
+
+  const sortData = (key: string) => {
+   let direction = 'asc';
+   if (sortConfig.key === key && sortConfig.direction === 'asc') {
+     direction = 'desc';
+   }
+   setSortConfig({ key, direction });
+
+   const sortedData = [...(activeTab === 'liquidity' ? pools : savers)].sort((a, b) => {
```

Comment:
not a huge fan of sorting on mixed types where a and b aren't known ahead of time.